### PR TITLE
Make FluentUI allow app extension API only

### PR DESCRIPTION
**Pull request by kkj1996**
### Platforms Impacted
- [✅] iOS
- [❌] macOS

### Description of changes
Make FluentUI allow app extension API only.
#### Why creates this PR?
There are two reasons:
* It is better choice to keep this project app extension API safe.
* We want to use Fluent on our app's framework and the framework is already app extension API safe.
### Verification
This changes has no build error or warning.
### Pull request checklist

This PR has considered:
- [✅]] Light and Dark appearances
- [✅]] VoiceOver and Keyboard Accessibility
- [✅]] Internationalization and Right to Left layouts
- [✅]] Different resolutions (1x, 2x, 3x)
- [✅]] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/448)

### CODE REVIEW

File: ios/FluentUI.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1625,6 +1625,7 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = 53FC90F825673626008A06FD /* FluentUI_framework.xcconfig */;
 			buildSettings = {
+				APPLICATION_EXTENSION_API_ONLY = YES;
```
Comment (anandrajeswaran) [inline ios/FluentUI.xcodeproj/project.pbxproj:1628]: Can we add to FluentUI_framework.xcconfig instead so we don't have to duplicate for each config in the very difficult-to-read .pbxproj file?

File: ios/FluentUI/People Picker/ContactCollectionView.swift [hunk 2]
```diff
@@ -111,7 +111,7 @@ open class ContactCollectionView: UICollectionView {
     }
 
     @objc private func updateHeightConstraint() {
-        let height = UIApplication.shared.preferredContentSizeCategory.contactHeight(size: size.contactViewSize) + 2 * Constants.verticalInset
+        let height = UIScreen.main.traitCollection.preferredContentSizeCategory.contactHeight(size: size.contactViewSize) + 2 * Constants.verticalInset
```
Comment (rdeassis) [inline ios/FluentUI/People Picker/ContactCollectionView.swift:114]: Can you explain in the PR description why this change is needed?

**Review by rdeassis — COMMENTED** [review]
Thank you for your contribution to FluentUI.

Could you please elaborate on the PR description with more details about the reason for these changes? (the code needs to be compatible with iOS extensions, etc...)

Same thing for the verification: I believe you have an iOS extension using FluentUI and you validated the changes in it. It'll be great to have those details here as well.

File: ios/FluentUI/People Picker/ContactCollectionView.swift (hunk 2 above)
Comment (sophialee0416) [reply ios/FluentUI/People Picker/ContactCollectionView.swift:114]: Either way we might not want to use UIScreen... #77

File: ios/FluentUI.xcodeproj/project.pbxproj (hunk 1 above)
Comment (kkj1996) [reply ios/FluentUI.xcodeproj/project.pbxproj:1628]: Great suggestion. I'll do it.

File: ios/FluentUI/People Picker/ContactCollectionView.swift (hunk 2 above)
Comment (kkj1996) [reply ios/FluentUI/People Picker/ContactCollectionView.swift]: Thanks for your review. I've updated.

File: ios/FluentUI/Extensions/UIView+Extensions.swift [hunk 3]
```diff
@@ -6,6 +6,11 @@
 import UIKit
 
 extension UIView {
+
+    var viewScreen: UIScreen {
+        return window?.screen ?? .main
```
Comment (rdeassis) [inline ios/FluentUI/Extensions/UIView+Extensions.swift:11]: The UIWindows' screen property is deprecated:
https://developer.apple.com/documentation/uikit/uiwindow/1621597-screen

We shouldn't be using it.
Please remove this computed property.

File: ios/FluentUI/People Picker/ContactCollectionView.swift [hunk 4]
```diff
@@ -111,7 +111,7 @@ open class ContactCollectionView: UICollectionView {
     }
 
     @objc private func updateHeightConstraint() {
-        let height = UIApplication.shared.preferredContentSizeCategory.contactHeight(size: size.contactViewSize) + 2 * Constants.verticalInset
+        let height = viewScreen.traitCollection.preferredContentSizeCategory.contactHeight(size: size.contactViewSize) + 2 * Constants.verticalInset
```
Comment (rdeassis) [inline ios/FluentUI/People Picker/ContactCollectionView.swift:114]: We should use the UIView's traitCollection instead. (here and in the other changes too)

**Review by rdeassis — CHANGES_REQUESTED** [review]
Can you please elaborate with more details on the reasons for this change so we can benefit from the documentation point of view?

I've also requested some changes in the implementation as well.
We should be using the UIView's trait collection instead of retrieving it from the deprecated UIWindow's screen property.

Thanks.

File: ios/FluentUI/People Picker/ContactCollectionView.swift [hunk 5]
```diff
@@ -111,7 +111,7 @@ open class ContactCollectionView: UICollectionView {
     }
 
     @objc private func updateHeightConstraint() {
-        let height = UIApplication.shared.preferredContentSizeCategory.contactHeight(size: size.contactViewSize) + 2 * Constants.verticalInset
+        let height = self.traitCollection.preferredContentSizeCategory.contactHeight(size: size.contactViewSize) + 2 * Constants.verticalInset
```
Comment (rdeassis) [inline ios/FluentUI/People Picker/ContactCollectionView.swift:114]: nit: can you remove the redundant "self." from these calls?

File: ios/FluentUI/People Picker/ContactView.swift [hunk 6]
```diff
@@ -41,7 +41,7 @@ open class ContactView: UIControl {
 
         var width: CGFloat {
             var width = avatarSize.size.width
-            switch UIApplication.shared.preferredContentSizeCategory {
+            switch UIView().traitCollection.preferredContentSizeCategory {
```
Comment (rdeassis) [inline ios/FluentUI/People Picker/ContactView.swift:44]: Using [UITraitCollection.current](https://developer.apple.com/documentation/uikit/uitraitcollection/3238080-current) seems like a better option here.

Your branch seems a bit outdated. We've already bumped the target deployment version to iOS13.

File: ios/FluentUI/Table View/TableViewCellFileAccessoryView.swift [hunk 7]
```diff
@@ -379,7 +381,7 @@ open class TableViewCellFileAccessoryView: UIView {
         let availableColumnSpace = cellSize.width - reservedCellSpace - width
         let columnMinWidth = Constants.columnMinWidth + Constants.columnSpacing
         let canShowFirstColumn = cellSize.width >= Constants.layoutBreakPoints[4] && availableColumnSpace > columnMinWidth
-        let canShowSecondColumn = cellSize.width >= Constants.layoutBreakPoints[5] && availableColumnSpace > 2 * columnMinWidth && UIApplication.shared.preferredContentSizeCategory.canShowSharedStatus
+        let canShowSecondColumn = cellSize.width >= Constants.layoutBreakPoints[5] && availableColumnSpace > 2 * columnMinWidth && self.traitCollection.preferredContentSizeCategory.canShowSharedStatus
```
Comment (rdeassis) [inline ios/FluentUI/Table View/TableViewCellFileAccessoryView.swift:384]: nit: remove "self."?

Comment (kkj1996) [reply ios/FluentUI/Table View/TableViewCellFileAccessoryView.swift]: Thanks!

File: ios/FluentUI/People Picker/ContactView.swift (hunk 6 above)
Comment (kkj1996) [reply ios/FluentUI/People Picker/ContactView.swift]: I've aligned to main branch.

File: ios/FluentUI/People Picker/ContactCollectionView.swift (hunk 5 above)
Comment (kkj1996) [reply ios/FluentUI/People Picker/ContactCollectionView.swift]: Ok.

File: ios/FluentUI/People Picker/ContactView.swift [hunk 8]
```diff
@@ -200,7 +200,7 @@ open class ContactView: UIControl {
     }
 
     private func updateLabelHeight() {
-        let contactHeight = UIApplication.shared.preferredContentSizeCategory.contactHeight(size: size)
+        let contactHeight = self.traitCollection.preferredContentSizeCategory.contactHeight(size: size)
```
Comment (rdeassis) [inline ios/FluentUI/People Picker/ContactView.swift:203]: Still missing this one:

nit: remove "self."?

**Review by rdeassis — APPROVED** [review]
